topics/functions-intro.md

```diff
@@ -1,21 +1,24 @@
 ---
 title: "Functions"
 description: >
-   Scripting with Redis OSS 7 and beyond
+   Scripting with functions stored on the server
 ---
 
-Valkey Functions is an API for managing code to be executed on the server. This feature, which became available in Redis OSS 7, supersedes the use of [EVAL](eval-intro.md) in prior versions of Valkey.
+Valkey Functions is an API for managing code to be executed on the server.
+This feature is as a complement to [EVAL scripts](eval-intro.md).
 
-## Prologue (or, what's wrong with Eval Scripts?)
+## What's wrong with EVAL?
 
-Prior versions of Valkey made scripting available only via the `EVAL` command, which allows a Lua script to be sent for execution by the server.
-The core use cases for [Eval Scripts](eval-intro.md) is executing part of your application logic inside Valkey, efficiently and atomically.
+There's nothing wrong with `EVAL`, but there are some differences between EVAL scripts and Functions.
+With the [`EVAL`](../commands/eval.md) command, scripts are sent to the server for immediate execution.
+The core use cases for `EVAL` scripts is executing part of your application logic inside Valkey, efficiently and atomically.
 Such script can perform conditional updates across multiple keys, possibly combining several different data types.
 
 Using `EVAL` requires that the application sends the entire script for execution every time.
-Because this results in network and script compilation overheads, Valkey provides an optimization in the form of the `EVALSHA` command. By first calling `SCRIPT LOAD` to obtain the script's SHA1, the application can invoke it repeatedly afterward with its digest alone.
+Because this results in network and script compilation overheads, Valkey provides an optimization in the form of the [`EVALSHA`](../commands/evalsha.md) command.
```

Comment:
This is more of a question for the style guide (which we still need to make).

Should _every_ instance of a command need to be linked or only the first?

Reply:
Only the first? WDYT?

Here, I added it only on the first occurrence (at least that was the intention).

On another website where this content has been used, every occurrence of all commands were automatically linked. If we want that, then I think we should do that too. I'm not sure I like that though. Do you?

Reply:
I'm honestly split @zuiderkwast. I thought that was what you were going with here. I do think we need to start linking topics and command reference in some way.

Pros for link-on-first-command only:
1. Less obnoxious to do manually.
2. Less likely of a clash between a arguments that _look_ like commands (e.g. argument `GET` in BITFIELD linked to command `GET`).
3. Overly linked pages have some negative SEO impact as it dilutes the flow between pages

Cons for link-on-first-command only:
1. Entry point for documentation might not be linear, so if someone links to a header, the actual link is _above_ the header.
2. Moving text around could potentially change whichever is the "first" link

(Also: I could easily automate this from the website using JS but it doesn't help us SEO wise and it doesn't help man pages)

Reply:
So you want automatic linking? In that case, it should be done in the pre-processing phase (zola), not in the browser.

The bang-syntax is all there just to indicate that it shouldn't be auto-linked, for example the `GET` option to the `SET` command is marked as `!GET`, which we'd remove in the pre-processing phase. If we're going this way, then we should **not** delete the bangs.

For man pages, auto-linking doesn't make much sense, since there are no links in man pages. It's better to list them at the bottom of the page in a SEE ALSO section. I could do that in the preprocessing too.